x-pack/legacy/plugins/ml/public/jobs/new_job_new/common/job_creator/job_creator.ts

```diff
@@ -245,6 +245,70 @@ export class JobCreator {
     }
   }
 
+  public set summaryCountFieldName(fieldName: string | null) {
+    if (fieldName !== null) {
+      this._job_config.analysis_config.summary_count_field_name = fieldName;
+    } else {
+      delete this._job_config.analysis_config.summary_count_field_name;
+    }
+  }
+
+  public get summaryCountFieldName(): string | null {
+    return this._job_config.analysis_config.summary_count_field_name || null;
+  }
+
+  public set categorizationFieldName(fieldName: string | null) {
+    if (fieldName !== null) {
+      this._job_config.analysis_config.categorization_field_name = fieldName;
+    } else {
+      delete this._job_config.analysis_config.categorization_field_name;
+    }
+  }
+
+  public get categorizationFieldName(): string | null {
+    return this._job_config.analysis_config.categorization_field_name || null;
+  }
+
+  public addCategorizationFilter(filter: string) {
+    if (this._job_config.analysis_config.categorization_filters === undefined) {
+      this._job_config.analysis_config.categorization_filters = [];
+    }
+
+    const filters = this._job_config.analysis_config.categorization_filters;
+    if (filters.includes(filter) === false) {
```

Comment:
i guess it's a matter of taste. i prefer writing boolean checks this way rather than say 
`if (!filters.includes(filter)) {`
as i think it is more obvious. `!` chars can be missed.